### ISSUE TITLE
Enhancement - RegKey

### ISSUE THREAD

**Issue by mrlesmithjr** [PowerDNS/dnsdist-ansible#2]
We need a way to generate the client key during provisioning with Ansible. It appears from everything that I have attempted that when the key is generated using scripting/Ansible that it is not able to be captured. Or is there a compelling reason to not allow such? Below are the things I have attempted.

```
- name: config_dnsdist | Generating New setKey
  shell: dnsdist -e "makeKey()" > /etc/dnsdist/setKey

- name: config_dnsdist | Generating New setKey
  shell: dnsdist -e "makeKey()"
  register: "setKey"
```

If I run the following from CLI on DNSDist host it works just fine but still cannot get the key to be captured.

```
sudo dnsdist -e "makeKey()"
```

```
setKey("JlUKwlbqfMaUJ+6GN9+KbFw8Gfw9xTQF7V3KcOK7drs=")
```

Posting here even though my Ansible role is not exactly the same as this role but it could benefit everyone if this is possible.

**Comment by pieterlexis** [comment]
I think you're running to the issue fixed in https://github.com/PowerDNS/pdns/pull/4553. This is fixed in the current master and will be part of dnsdist 1.1.0 (which will be released within a week or two). In the meantime, you can use a static key instead and provision it using `dnsdist_setkey`

As for the feature request, I don't think generating the key on the spot is something that is useful. The key is a shared secret that should also be known on the client's end (aka the operators machine).

**Comment by mrlesmithjr** [comment]
@pieterlexis Thanks for the feedback. Understood on the client's end but this could also be something configured using Ansible at the time of key generation. If we collect that client key we could write it out to the host_vars/hostname.yml file and/or anywhere else. But again understood and if this is fixed in the upcoming release we can definitely use this method in our own specific use case.

Much appreciated.

**Comment by atosatto** [comment]
This is now fixed in master thanks to @Godwottery's PR #23.
